spec/custom_facts/util/normalization_spec.rb

```diff
@@ -126,4 +134,16 @@ def utf8(str)
       end
     end
   end
+
+  describe 'encoding strings' do
+    it 'correctly matches encoding for UTF-16LE byte sequence' do
+      str = 'Hello World!'.encode(Encoding::UTF_16LE)
+      expect(str.encoding).to eq(Encoding::UTF_16LE)
+    end
+
+    it 'correctly matches encoding for string that contains binary data' do
+      str = 'Hello World!'.encode(Encoding::BINARY) + 'Hello again!'
+      expect(str.encoding).to eq(Encoding::BINARY)
+    end
+  end
```

Comment:
Wasn't sure about this part, were you wanting to pass a non-UTF8 string to `normalize` and verify you get back UTF8, something like?

```ruby
str = 'Hello World!'.encode(Encoding::UTF_16LE)
expect(normallization.normalize(str.encoding)).to eq(Encoding::UTF_8)
```

Reply:
Ah I was trying to make tests for these examples in the ticket:

- String contains a valid UTF-16LE byte sequence, but the string's "encoding" method returns UTF-8 (Windows Registry)
- String contains binary data, but "encoding" returns UTF-8 (EC2 userdata)